### PR TITLE
Gpu init

### PR DESCRIPTION
**Pull request by liruipeng**
This PR adds a lazy initialization feature for GPU build. The GPU initialization is moved into  `HYPRE_DeviceInitialize` and out of `HYPRE_Initialize`. So, the old `HYPRE_Initialize` is equivalent to
```
HYPRE_Initialize();
HYPRE_DeviceInitialize();
```
in this PR.

- [x] tux
- [x] lassen
- [x] tioga
- [x] sunspot

### CODE REVIEW

File: src/utilities/general.c [hunk 1]
```diff
@@ -236,35 +269,19 @@ hypre_GetDeviceLastError(void)
...
-   {
-      _hypre_handle = hypre_HandleCreate();
-   }
+   hypre_Handle *handle = hypre_handle();
 
 #if defined(HYPRE_USING_GPU) || defined(HYPRE_USING_DEVICE_OPENMP)
 #if defined(HYPRE_USING_SYCL)
    /* WM: note that for sycl, we need to do device setup again if reinitializing */
-   if (!HYPRE_Initialized())
+   //if (!HYPRE_Initialized())
```
Comment (liruipeng) [inline src/utilities/general.c]: - [x] Need to talk about this logics ....

Comment (victorapm) [reply src/utilities/general.c]: In my opinion, these lines can be removed:
https://github.com/hypre-space/hypre/blob/e6785d07e78485f6f5b6b79b5fe6a0dd50efcf88/src/utilities/general.c#L277-L285

Comment (liruipeng) [reply src/utilities/general.c]: @victorapm @waynemitchell K, let's talk about it in an SLS meeting. I want to confirm what we want to support for Init/ReInit, etc. And in particular with sycl. Thanks

**Review by victorapm — APPROVED** [review]

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Comment by liruipeng** [comment]
I am still waiting for tioga back in service so I can test on AMD. 

@waynemitchell Do you have more comments on this PR?

**Review by waynemitchell — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**